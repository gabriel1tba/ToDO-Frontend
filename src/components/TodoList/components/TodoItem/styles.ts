import styled from 'styled-components';

export const Wrapper = styled.li`
  display: flex;
  justify-content: space-between;
  align-items: center;

  margin: 15px 5px -5px;

  padding-bottom: 14px;

  position: relative;

  &:before {
    content: '';
    position: absolute;
    left: 0;
    bottom: 5px;
    width: calc(100vw - 765px);
    border: 0.5px solid rgba(0, 0, 0, 0.125);
  }

  > input[type='checkbox'] {
    width: 17px !important;
    height: 17px !important;
    margin: 5px;
    appearance: none;
    box-shadow: none;
    font-size: 2em;
    border: solid 2px #2ecc71;

    &:checked {
      background-color: #2ecc71;

      display: inline-block;

      padding-top: -2px;
      padding-left: 3px;
      &:after {
        content: '';
        display: block;

        width: 4.5px;
        height: 9px;

        border: solid #fff;
        border-width: 0 2px 2px 0;

        transform: rotate(45deg);
      }
    }
  }

  > a {
    cursor: pointer;
    width: calc(100vw - 860px);
  }

  > div {
    margin-right: 350px;

    button {
      background: transparent;
      border: none;
      svg {
        cursor: pointer;
        margin-left: 5px;

        transition: transform 0.3s;

        &:hover {
          transform: scale(1.2);
        }
      }
    }
  }
`;
