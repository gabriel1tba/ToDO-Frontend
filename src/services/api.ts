import axios from 'axios';

const api = axios.create({
  baseURL: `${process.env.REACT_APP_API_HOST}:${process.env.REACT_APP_API_PORT}`,
  headers: {
    'Content-Type': 'application/json;charset=UTF-8',
    Accept: 'application/json',
  },
});

export default api;
