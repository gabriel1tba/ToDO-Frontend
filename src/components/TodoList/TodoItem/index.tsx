import { useEffect, useState } from 'react';
import { TbTrash, TbEdit } from 'react-icons/tb';

import * as S from './styles';

import { useToast, useTodos, useToggle } from 'hooks';
import { ActionType } from 'context/todos/actions';

import TodoService from 'services/TodoService';

import Alert from 'components/Alert';
import Modal from 'components/Modal';
import CheckBox from 'components/CheckBox';

import { ITodoItem } from 'components/TodoList/interfaces';

import EditTodo from '../Forms/EditTodo';

const TodoItem = ({ todo }: ITodoItem) => {
  const { addToast } = useToast();
  const { todoDispatch } = useTodos();

  const [openModal, handleToggleModal] = useToggle();
  const [openAlert, handleToggleAlert] = useToggle();

  const [, setDidMount] = useState(false);

  useEffect(() => {
    setDidMount(true);
    return () => setDidMount(false);
  }, []);

  const handleCompletedTodo = async (checked: boolean) => {
    try {
      const { data } = await TodoService.updateTodo({
        id: todo.id,
        title: todo.title,
        completed: checked,
      });

      todoDispatch({ type: ActionType.UpdateTodo, payload: data });
    } catch {
      addToast({
        type: 'error',
        title: 'Erro ao fazer alteração.',
        description: 'Um erro ocorreu ao marcar tarefa como completa.',
        secondsDuration: 8,
      });
    }
  };

  return (
    <>
      <S.Wrapper isCompleted={todo.completed}>
        <CheckBox
          id="completed"
          border="round"
          name="completed"
          checked={todo.completed}
          onChange={(event) => {
            handleCompletedTodo(event.target.checked);
          }}
        />

        <p>{todo.title}</p>

        <div>
          <TbEdit onClick={handleToggleModal} />
          <TbTrash onClick={handleToggleAlert} />
        </div>
      </S.Wrapper>

      <Modal
        title="Editar tarefa"
        onCloseModal={handleToggleModal}
        open={openModal}
      >
        <EditTodo todo={todo} onCloseModal={handleToggleModal} />
      </Modal>

      <Alert
        isOpen={openAlert}
        title="Excluir tarefa"
        description="Tem certeza que deseja excluir essa tarefa?"
        onClose={handleToggleAlert}
        onConfirm={handleToggleAlert}
      />
    </>
  );
};

export default TodoItem;
