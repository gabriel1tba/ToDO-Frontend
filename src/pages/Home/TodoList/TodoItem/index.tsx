import { useState, useEffect, useCallback } from 'react';
import { TbTrash, TbEdit } from 'react-icons/tb';

import TodoService from 'services/TodoService';

import { useToast, useToggle } from 'hooks';
import { useTodos } from 'pages/Home';
import { updateTodoAction, deleteTodoAction } from 'pages/Home/utils/actions';

import Alert from 'components/Alert';
import Modal from 'components/Modal';
import CheckBox from 'components/CheckBox';
import EditTodo from '../../Forms/EditTodo';

import * as S from './styles';

import { ITodoItem } from '../../interfaces';

const TodoItem = ({ todo }: ITodoItem) => {
  const [, setDidMount] = useState(false);

  useEffect(() => {
    setDidMount(true);
    return () => setDidMount(false);
  }, []);

  const { addToast } = useToast();
  const { dispatchTodos } = useTodos();

  const [openModal, handleToggleModal] = useToggle();
  const [openAlert, handleToggleAlert] = useToggle();
  const [loadingAlert, toggleLoadingAlert] = useToggle();

  const handleCompletedTodo = async (checked: boolean) => {
    try {
      const { data } = await TodoService.updateTodo({
        id: todo.id,
        title: todo.title,
        completed: checked,
      });

      dispatchTodos(updateTodoAction(data));
    } catch {
      addToast({
        type: 'error',
        title: 'Erro ao fazer alteração.',
        description: 'Um erro ocorreu ao marcar tarefa como completa.',
        secondsDuration: 8,
      });
    }
  };

  const handleDeleteTodoAcdeleteTodoAction = useCallback(async () => {
    toggleLoadingAlert();

    try {
      await TodoService.deleteTodo({
        id: todo.id,
      });

      dispatchTodos(deleteTodoAction(todo.id));

      addToast({
        type: 'success',
        title: 'Tarefa foi removida com sucesso.',
        secondsDuration: 8,
      });
    } catch {
      addToast({
        type: 'error',
        title: 'Erro ao remover tarefa!',
        secondsDuration: 8,
      });
    } finally {
      handleToggleAlert();
      toggleLoadingAlert();
    }
  }, [toggleLoadingAlert, todo, dispatchTodos, addToast, handleToggleAlert]);

  return (
    <>
      <S.Wrapper isCompleted={!!todo.completed}>
        <CheckBox
          id="completed"
          border="round"
          name="completed"
          checked={todo.completed}
          onChange={(event) => {
            handleCompletedTodo(event.target.checked);
          }}
        />

        <p>{todo.title}</p>

        <div>
          <button onClick={handleToggleModal} data-testid="edit-todo">
            <TbEdit />
          </button>
          <button onClick={handleToggleAlert} data-testid="delete-todo">
            <TbTrash />
          </button>
        </div>
      </S.Wrapper>

      <Modal
        title="Editar tarefa"
        onCloseModal={handleToggleModal}
        open={openModal}
      >
        <EditTodo todo={todo} onCloseModal={handleToggleModal} />
      </Modal>

      <Alert
        isOpen={openAlert}
        isLoading={loadingAlert}
        title="Excluir tarefa"
        description="Tem certeza que deseja excluir essa tarefa?"
        onClose={handleToggleAlert}
        onConfirm={handleDeleteTodoAcdeleteTodoAction}
      />
    </>
  );
};

export default TodoItem;
