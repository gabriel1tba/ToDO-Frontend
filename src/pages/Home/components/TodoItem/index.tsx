import { useState, useEffect } from 'react';
import { TbTrash, TbEdit } from 'react-icons/tb';

import TodoService from 'services/TodoService';

import { useToastContext } from 'hooks';
import { useTodosContext } from 'pages/Home';
import { updateTodoAction } from 'pages/Home/utils/actions';

import CheckBox from 'components/CheckBox';

import * as S from './styles';

import { ITodoItem } from '../../interfaces';

const TodoItem = ({ todo, onEditTodoModal, onOpenAlert }: ITodoItem) => {
  const [, setDidMount] = useState(false);

  useEffect(() => {
    setDidMount(true);

    return () => setDidMount(false);
  }, []);

  const { addToast } = useToastContext();
  const { dispatchTodos } = useTodosContext();

  const handleCompleteTodo = async (checked: boolean) => {
    try {
      const { data } = await TodoService.updateTodo({
        id: todo.id,
        title: todo.title,
        completed: checked,
      });

      dispatchTodos(updateTodoAction(data));
    } catch {
      addToast({
        type: 'error',
        title: 'Erro ao fazer alteração.',
        description: 'Um erro ocorreu ao marcar tarefa como completa.',
        secondsDuration: 8,
      });
    }
  };

  return (
    <S.Wrapper isCompleted={!!todo.completed}>
      <button
        onClick={() => {
          handleCompleteTodo(!todo.completed);
        }}
      >
        <CheckBox
          id="completed"
          border="round"
          name="completed"
          checked={todo.completed}
          onChange={() => null}
        />

        <p>{todo.title}</p>
      </button>

      <div>
        <button onClick={() => onEditTodoModal?.(todo)} data-testid="edit-todo">
          <TbEdit />
        </button>
        <button onClick={() => onOpenAlert?.(todo)} data-testid="delete-todo">
          <TbTrash />
        </button>
      </div>
    </S.Wrapper>
  );
};

export default TodoItem;
