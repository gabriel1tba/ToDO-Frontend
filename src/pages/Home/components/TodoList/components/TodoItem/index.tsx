import { useState, useEffect, useCallback } from 'react';
import { TbTrash, TbEdit } from 'react-icons/tb';

import TodoService from 'services/TodoService';

import { useToastContext, useToggle } from 'hooks';
import { useTodosContext } from 'pages/Home';
import { updateTodoAction, deleteTodoAction } from 'pages/Home/utils/actions';

import Alert from 'components/Alert';
import Modal from 'components/Modal';
import CheckBox from 'components/CheckBox';
import EditTodo from '../../../Forms/EditTodo';

import * as S from './styles';

import { ITodoItem } from '../../../../interfaces';

const TodoItem = ({ todo }: ITodoItem) => {
  const [openAlert, setOpenAlert] = useState({
    isOpen: false,
    todoTitle: '',
  });

  const [, setDidMount] = useState(false);

  useEffect(() => {
    setDidMount(true);
    return () => setDidMount(false);
  }, []);

  const { addToast } = useToastContext();
  const { dispatchTodos } = useTodosContext();

  const [openModal, handleToggleModal] = useToggle();
  const [loadingAlert, handleToggleLoading] = useToggle();

  const handleCompleteTodo = async (checked: boolean) => {
    try {
      const { data } = await TodoService.updateTodo({
        id: todo.id,
        title: todo.title,
        completed: checked,
      });

      dispatchTodos(updateTodoAction(data));
    } catch {
      addToast({
        type: 'error',
        title: 'Erro ao fazer alteração.',
        description: 'Um erro ocorreu ao marcar tarefa como completa.',
        secondsDuration: 8,
      });
    }
  };

  const handleDeleteTodo = useCallback(async () => {
    handleToggleLoading();

    try {
      await TodoService.deleteTodo({
        id: todo.id,
      });

      dispatchTodos(deleteTodoAction(todo.id));

      addToast({
        type: 'success',
        title: 'Tarefa foi removida com sucesso.',
        secondsDuration: 8,
      });
    } catch {
      addToast({
        type: 'error',
        title: 'Erro ao remover tarefa!',
        secondsDuration: 8,
      });
    } finally {
      setOpenAlert((prevState) => ({
        isOpen: !prevState.isOpen,
        todoTitle: '',
      }));
      handleToggleLoading();
    }
  }, [addToast, dispatchTodos, todo.id, handleToggleLoading]);

  return (
    <>
      <S.Wrapper isCompleted={!!todo.completed}>
        <CheckBox
          id="completed"
          border="round"
          name="completed"
          checked={todo.completed}
          onChange={(event) => {
            handleCompleteTodo(event.target.checked);
          }}
        />

        <p>{todo.title}</p>

        <div>
          <button onClick={handleToggleModal} data-testid="edit-todo">
            <TbEdit />
          </button>
          <button
            onClick={() =>
              setOpenAlert((prevState) => ({
                isOpen: !prevState.isOpen,
                todoTitle: todo.title,
              }))
            }
            data-testid="delete-todo"
          >
            <TbTrash />
          </button>
        </div>
      </S.Wrapper>

      <Modal
        title="Editar tarefa"
        onCloseModal={handleToggleModal}
        open={openModal}
      >
        <EditTodo todo={todo} onCloseModal={handleToggleModal} />
      </Modal>

      <Alert
        isOpen={openAlert.isOpen}
        isLoading={loadingAlert}
        title={`Excluir tarefa "${openAlert.todoTitle}"?`}
        description="Tem certeza que deseja excluir essa tarefa?"
        confirmLabel="Sim, excluir"
        onCancel={() =>
          setOpenAlert((prevState) => ({
            isOpen: !prevState.isOpen,
            todoTitle: '',
          }))
        }
        onConfirm={handleDeleteTodo}
      />
    </>
  );
};

export default TodoItem;
