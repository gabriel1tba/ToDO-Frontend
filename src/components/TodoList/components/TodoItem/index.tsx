import { useEffect, useState } from 'react';
import { FaEdit } from 'react-icons/fa';
import { BsTrash } from 'react-icons/bs';
import { useTheme } from 'styled-components';

import * as S from './styles';

import { useToast, useTodos, useToggle } from 'hooks';

import TodoService from 'services/TodoService';

import Modal from 'components/Modal';

import ManageTodo from '../Forms/ManageTodo';

import { ActionType } from 'context/todos/actions';

import { ITodoItem } from 'components/TodoList/interfaces';

const TodoItem = ({ todo }: ITodoItem) => {
  const { addToast } = useToast();
  const { todoDispatch } = useTodos();
  const theme = useTheme();

  const [openModal, hadleToggleModal] = useToggle();

  const [showTodo, setShowTodo] = useState(false);
  const [editTodo, setEditTodo] = useState(false);

  // State to useEffect cleanup function
  const [, setDidMount] = useState(false);

  // useEffect cleanup function
  useEffect(() => {
    setDidMount(true);
    return () => setDidMount(false);
  }, []);

  const handleCompletedTodo = async (checked: boolean) => {
    try {
      const { data } = await TodoService.updateTodo({
        id: todo.id,
        title: todo.title,
        completed: checked,
      });

      todoDispatch({ type: ActionType.UpdateTodo, payload: data });
    } catch {
      addToast({
        type: 'error',
        title: 'Erro ao fazer alteração.',
        description: 'Um erro ocorreu ao marcar tarefa como completa.',
        secondsDuration: 8,
      });
    }
  };

  const handleViewTodo = () => {
    setShowTodo(true);
    setEditTodo(false);
    hadleToggleModal();
  };

  const handleEditTodo = () => {
    setShowTodo(false);
    setEditTodo(true);
    hadleToggleModal();
  };

  const handleDeleteTodo = () => {
    setShowTodo(false);
    setEditTodo(false);
    hadleToggleModal();
  };

  const setModalTitle = () => {
    return showTodo
      ? 'Detalhes da tarefa'
      : editTodo
      ? 'Editar tarefa'
      : 'Excluir tarefa';
  };

  return (
    <>
      <Modal
        title={setModalTitle()}
        onCloseModal={hadleToggleModal}
        open={openModal}
      >
        <ManageTodo
          todo={todo}
          editTodo={editTodo}
          showTodo={showTodo}
          onCloseModal={hadleToggleModal}
        />
      </Modal>

      <S.Wrapper>
        <input
          type="checkbox"
          name="completed"
          id="completed"
          defaultChecked={todo.completed}
          onChange={(event) => {
            handleCompletedTodo(event.target.checked);
          }}
        />

        <a
          style={{ textDecoration: todo.completed ? 'line-through' : 'none' }}
          onClick={handleViewTodo}
        >
          {todo.title}
        </a>

        <div>
          <button onClick={handleEditTodo}>
            <FaEdit
              size={20}
              color={theme.colors.warning.main}
              data-testid="edit-todo"
            />
          </button>
          <button onClick={handleDeleteTodo}>
            <BsTrash
              size={20}
              color={theme.colors.danger.main}
              data-testid="delete-todo"
            />
          </button>
        </div>
      </S.Wrapper>
    </>
  );
};

export default TodoItem;
