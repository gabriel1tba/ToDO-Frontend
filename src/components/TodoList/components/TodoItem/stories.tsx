import { Story, Meta } from '@storybook/react';

import TodoItem from '.';
import { ITodoItem } from '../../interfaces';

const todoMock = {
  id: '7b403461-2579-4635-af53-9c07f644f2ec',
  user_id: '6b14098c-7357-4fe5-9bba-24061c62bf1c',
  completed: false,
  title: 'Finalizar Curso CAR CRM',
  description: '',
  created_at: '2022-04-16T13:59:36.932Z',
  updated_at: '2022-04-16T13:59:40.339Z',
};

export default {
  title: 'Todo/Item',
  component: TodoItem,
} as Meta;

export const Default: Story<ITodoItem> = (args) => (
  <div style={{ width: 'fit-content', margin: 'auto' }}>
    <TodoItem todo={args.todo} />
  </div>
);

Default.args = {
  todo: todoMock,
};

export const Completed: Story<ITodoItem> = (args) => (
  <div style={{ width: 'fit-content', margin: 'auto' }}>
    <TodoItem todo={args.todo} />
  </div>
);

Completed.args = {
  todo: { ...todoMock, completed: true },
};
