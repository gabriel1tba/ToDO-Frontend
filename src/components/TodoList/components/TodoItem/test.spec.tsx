import userEvent from '@testing-library/user-event';
import { render, screen, waitFor } from 'utils/test-utils';
import MockAdapter from 'axios-mock-adapter';

import api from 'services/api';

const apiMock = new MockAdapter(api);

import TodoItem from '.';

const todoItemResponse = {
  id: 'todoId',
  user_id: 'userId',
  completed: false,
  title: 'Testar componente TodoItem',
  description: 'Testar em Todos os modos',
  created_at: '2021-10-10T07:12:55.137Z',
  updated_at: '2021-10-10T07:32:52.716Z',
};

describe('<TodoItem />', () => {
  it('should render the TodoItem correctly', () => {
    const { container } = render(<TodoItem todo={todoItemResponse} />);

    expect(screen.getByRole('checkbox')).not.toBeChecked();
    expect(screen.getByText(/testar componente todoitem/i)).toBeInTheDocument();
    expect(screen.getAllByRole('button')).toHaveLength(2);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the TodoItem with completed prop true', () => {
    render(<TodoItem todo={{ ...todoItemResponse, completed: true }} />);

    expect(screen.getByRole('checkbox')).toBeChecked();
    expect(screen.getByText(/testar componente todoitem/i)).toHaveStyle({
      textDecoration: 'line-through',
    });
  });

  it('should change completed status of whole to true', async () => {
    render(<TodoItem todo={todoItemResponse} />);

    expect(screen.getByRole('checkbox')).not.toBeChecked();

    userEvent.click(screen.getByRole('checkbox'));

    const responseCompleted = apiMock
      .onPatch('todos')
      .reply(() => [200, { ...todoItemResponse, completed: true }]);

    await waitFor(() => {
      expect(responseCompleted.history.patch.length).toBe(1);
    });

    await waitFor(() => {
      expect(responseCompleted.history.patch[0].data).toStrictEqual(
        JSON.stringify({
          id: 'todoId',
          completed: true,
          title: 'Testar componente TodoItem',
          description: 'Testar em Todos os modos',
        }),
      );
    });

    await waitFor(() => {
      expect(screen.getByRole('checkbox')).toBeChecked();
    });
  });
});
