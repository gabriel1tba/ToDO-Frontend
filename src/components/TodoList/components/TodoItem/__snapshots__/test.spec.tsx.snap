// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoItem /> should render the TodoItem correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 15px 5px -5px;
  padding-bottom: 14px;
  position: relative;
}

.c0:before {
  content: '';
  position: absolute;
  left: 0;
  bottom: 5px;
  width: calc(100vw - 765px);
  border: 0.5px solid rgba(0,0,0,0.125);
}

.c0 > input[type='checkbox'] {
  width: 17px !important;
  height: 17px !important;
  margin: 5px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  box-shadow: none;
  font-size: 2em;
  border: solid 2px #2ecc71;
}

.c0 > input[type='checkbox']:checked {
  background-color: #2ecc71;
  display: inline-block;
  padding-top: -2px;
  padding-left: 3px;
}

.c0 > input[type='checkbox']:checked:after {
  content: '';
  display: block;
  width: 4.5px;
  height: 9px;
  border: solid #fff;
  border-width: 0 2px 2px 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c0 > a {
  cursor: pointer;
  width: calc(100vw - 860px);
}

.c0 > div {
  margin-right: 350px;
}

.c0 > div button {
  background: transparent;
  border: none;
}

.c0 > div button svg {
  cursor: pointer;
  margin-left: 5px;
  -webkit-transition: -webkit-transform 0.3s;
  -webkit-transition: transform 0.3s;
  transition: transform 0.3s;
}

.c0 > div button svg:hover {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

<li
  class="c0"
>
  <input
    id="completed"
    name="completed"
    type="checkbox"
  />
  <a
    style="text-decoration: none;"
  >
    Testar componente TodoItem
  </a>
  <div>
    <button>
      <svg
        color="#ffc107"
        fill="currentColor"
        height="20"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(255, 193, 7);"
        viewBox="0 0 576 512"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
        />
      </svg>
    </button>
    <button>
      <svg
        color="#dc3545"
        fill="currentColor"
        height="20"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(220, 53, 69);"
        viewBox="0 0 16 16"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.5 5.5A.5.5 0 016 6v6a.5.5 0 01-1 0V6a.5.5 0 01.5-.5zm2.5 0a.5.5 0 01.5.5v6a.5.5 0 01-1 0V6a.5.5 0 01.5-.5zm3 .5a.5.5 0 00-1 0v6a.5.5 0 001 0V6z"
        />
        <path
          clip-rule="evenodd"
          d="M14.5 3a1 1 0 01-1 1H13v9a2 2 0 01-2 2H5a2 2 0 01-2-2V4h-.5a1 1 0 01-1-1V2a1 1 0 011-1H6a1 1 0 011-1h2a1 1 0 011 1h3.5a1 1 0 011 1v1zM4.118 4L4 4.059V13a1 1 0 001 1h6a1 1 0 001-1V4.059L11.882 4H4.118zM2.5 3V2h11v1h-11z"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
</li>
`;
